Makefile

```
@@ -0,0 +1,44 @@
+CARGO := $(shell command -v cargo 2> /dev/null)
+
+ifndef CARGO
+$(error "Cannot find cargo. Please install and try again!")
+endif
+
+all: help
+
+.PHONY: clean
+clean: ## Cleans up the project by removing the target directory.
+	@$(CARGO) clean
+
+.PHONY: lint
+lint: ## Runs Clippy to lint the codebase.
+	@$(CARGO) clippy --no-deps
+
+.PHONY: format
+format: ## Formats the codebase using rustfmt.
+	@$(CARGO) fmt
+
+.PHONY: check
+check: format lint ## Formats the codebase and then lints it.
+
+.PHONY: build
+build: ## Compiles the project.
+	@$(CARGO) build
+
+.PHONY: run
+run: ## Compiles and runs the project.
+	@$(CARGO) run
```

Perhaps we can separate this into two different targets:

```suggestion
.PHONY: debug
debug: ## Compiles and runs the project.
	@$(CARGO) run
	
.PHONY: run
run: ## Compiles and runs the project.
	@$(CARGO) run --release
```